programs/util.c

```diff
@@ -326,6 +326,35 @@ int UTIL_prepareFileList(const char *dirName, char** bufStart, size_t* pos, char
 
 #endif /* #ifdef _WIN32 */
 
+/* Check if the file is Compressed by comparing it with compressFileExtension list.
+YES => Skip the file (return 1)
+NO => return 0
+*/
+
+int UTIL_isCompressedFile(const char *inputName, const char *extensionList[])
+{
+  const char* ext = UTIL_getFileExtension(inputName);
+  if(strcmp(ext,""))
```

Comment:
This shouldn't be necessary?

Reply:
Removed the comments. The if statement when true (i.e. ext != "") will run the while loop for comparing the extension. If false, the extension comparison will be skipped since file will not have any extension (ext == "").

Reply:
This comparison, if I understand it correctly, is not necessary for correctness. You've added it as an optimization. Given that, I would rather not include it. This is not performance-sensitive code, and instead we care much more about ease-of-understanding and ease-of-verifying-correctness, which I think this harms.

Alternatively, if you feel strongly that this is a valuable optimization, it should be rephrased as:

```
if (!strcmp(ext, "")) {
  return 0;
}
```

Reply:
Agreed. Thanks for pointing this out. Addressed in latest commit.